synchronizer/l1_worker_etherman.go

```diff
@@ -108,32 +108,41 @@ type retrieveL1LastBlockResult struct {
 	block uint64
 }
 
-type worker struct {
+type workerEtherman struct {
 	mutex                sync.Mutex
 	etherman             EthermanInterface
 	status               ethermanStatusEnum
 	typeOfCurrentRequest typeOfRequest
 }
 
-func (w *worker) String() string {
+func (w *workerEtherman) String() string {
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
 	return fmt.Sprintf("status:%s req:%s", w.status.String(), w.typeOfCurrentRequest.String())
 }
 
-func newWorker(etherman EthermanInterface) *worker {
-	return &worker{etherman: etherman, status: ethermanIdle}
+func newWorker(etherman EthermanInterface) *workerEtherman {
+	return &workerEtherman{etherman: etherman, status: ethermanIdle}
 }
 
-func (w *worker) asyncRequestRollupInfoByBlockRange(ctx context.Context, ch chan responseRollupInfoByBlockRange, wg *sync.WaitGroup, blockRange blockRange) error {
+func (w *workerEtherman) asyncRequestRollupInfoByBlockRange(ctx context.Context, cancelCtx context.CancelFunc, ch chan responseRollupInfoByBlockRange, wg *sync.WaitGroup, blockRange blockRange) error {
```

Comment:
Maybe we can pass a struct:
```
type ctxWithCancel struct {
   ctx       context.Context
   cancelCtx context.CancelFunc
}
```

Reply:
I like very much the idea, I'll do the changes